node/src/components/block_proposer.rs

```diff
@@ -83,39 +89,29 @@ impl BlockProposer {
         registry: Registry,
         effect_builder: EffectBuilder<REv>,
         next_finalized_block: BlockHeight,
+        chainspec: &Chainspec,
     ) -> Result<(Self, Effects<Event>), prometheus::Error>
     where
         REv: From<Event> + From<StorageRequest> + From<StateStoreRequest> + Send + 'static,
     {
-        // Note: Version is currently not honored by the storage component, so we just hardcode
-        // 1.0.0.
+        // load the state from storage or use a fresh instance if loading fails.
+        let state_key = deploy_sets::create_storage_key(chainspec);
         let effects = async move {
-            let chainspec = effect_builder
-                .get_chainspec(Version::new(1, 0, 0))
-                .await
-                // Note: Currently the storage component will always return a chainspec, however the
-                // interface has not kept up with this yet.
-                .expect("chainspec should be infallible");
-
-            // With the chainspec, we can now load the state from storage or use a fresh instance if
-            // loading fails.
-            let key = deploy_sets::create_storage_key(&chainspec);
-            let sets = effect_builder
-                .load_state(key.into())
+            effect_builder
+                .load_state(state_key.into())
                 .await
-                .unwrap_or_default();
-
-            (chainspec, sets)
+                .unwrap_or_default()
         }
-        .event(move |(chainspec, sets)| Event::Loaded {
-            chainspec,
+        .event(move |sets| Event::Loaded {
             sets,
             next_finalized_block,
         });
 
         let block_proposer = BlockProposer {
             state: BlockProposerState::Initializing {
                 pending: Vec::new(),
+                state_key: deploy_sets::create_storage_key(chainspec),
```

Comment:
Is that the same as `state_key` in line 98?

Reply:
Yes - I'll clone it above to make this more obvious.